lightly_studio/src/lightly_studio/core/add_videos.py

```diff
@@ -0,0 +1,213 @@
+from tqdm import tqdm
+
+from lightly_studio.core.logging import _LoadingLoggingContext, _log_loading_results
+from lightly_studio.models.video import VideoCreate, VideoFrameCreate
+from lightly_studio.resolvers import (
+    dataset_resolver,
+    sample_resolver,
+    video_frame_resolver,
+    video_resolver,
+)
+
+_DEFAULT_VIDEO_CHANNEL = 0
+SAMPLE_BATCH_SIZE = 32  # Number of samples to process in a single batch
+
+
+def load_into_dataset_from_paths(
+    session: Session,
+    dataset_id: UUID,
+    video_paths: Iterable[str],
+    video_channel: int = _DEFAULT_VIDEO_CHANNEL,
+    fps: float | None = None,
+) -> list[UUID]:
+    """Load video frames from file paths into the dataset using PyAV.
+
+    Args:
+        session: The database session.
+        dataset_id: The ID of the dataset to load video frames into.
+        video_paths: An iterable of file paths to the videos to load.
+        video_channel: The video channel from which frames are loaded.
+        fps: Optional FPS value to control frame extraction. If provided, only frames
+            at the specified FPS intervals will be extracted. If None, all frames
+            will be extracted.
+
+    Returns:
+        A list of UUIDs of the created samples.
+    """
+    created_sample_ids: list[UUID] = []
+    file_paths_new, file_paths_exist = video_resolver.filter_new_paths(
+        session=session, file_paths_abs=list(video_paths)
+    )
+    video_logging_context = _LoadingLoggingContext(
+        n_samples_to_be_inserted=sum(1 for _ in video_paths),
+        n_samples_before_loading=sample_resolver.count_by_dataset_id(
+            session=session, dataset_id=dataset_id
+        ),
+    )
+    video_logging_context.update_example_paths(file_paths_exist)
+    # Get the video frames dataset ID
+    video_frames_dataset_id = dataset_resolver.get_or_create_video_frame_child(
+        session=session, dataset_id=dataset_id
+    )
+    for video_path in tqdm(
+        file_paths_new,
+        desc="Loading videos",
+        unit=" videos",
+    ):
+        try:
+            # Open video and extract metadata
+            fs, fs_path = fsspec.core.url_to_fs(video_path)
+            content = fs.cat_file(fs_path)
+            video_buffer = BytesIO(content)
+            # When opening a file for reading, container is InputContainer
+            video_container: av.container.InputContainer = container.open(
+                video_buffer,
+            )  # type: ignore
+            video_stream = video_container.streams.video[video_channel]
+
+            # Get video metadata
+            framerate = float(video_stream.average_rate) if video_stream.average_rate else 0.0
+            video_width = video_stream.width if video_stream.width else 0
+            video_height = video_stream.height if video_stream.height else 0
+            if video_stream.duration and video_stream.time_base:
+                video_duration = float(video_stream.duration * video_stream.time_base)
```

Comment:
We have to change the fields we store in the VideoTable and store duration_pts and time_base. The issue is that without it we would be doing floating point arithmetic which might cause issues since it is not precise.

That's something for a follow-up PR.